vllm/engine/async_llm_engine.py

```diff
@@ -106,9 +105,9 @@ async def generator(
         self
     ) -> AsyncGenerator[Union[RequestOutput, EmbeddingRequestOutput], None]:
         try:
-            while not self._finished:
+            while True:
```

Comment:
Am I understanding right

- we have a background loop putting things onto the queue ... we could put 10 items onto the queue
- the background loop will call stream.finish() at some point
- this function gets items from the queue ... so it may only get 8 items off before finish() is called
- so instead, we should only stop based on the results of the queue

Reply:
@robertgshaw2-neuralmagic yes that's right.. the "background loop" here is the main model loop producing the tokens.

This function's loop is driven by how fast the AsyncLLMEngine is consuming.

Reply:
makes sense